app/components/conceptAdd/conceptAdd.controller.specj.js

```diff
@@ -0,0 +1,112 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+'use strict';
+
+/* jasmine specs for controllers go here */
+describe('Concept dictionary controllers', function() {
+
+    beforeEach(function(){
+        jasmine.addMatchers({
+            toEqualData: function(util, customEqualityTesters) {
+                return {
+                    compare: function(actual, expected) {
+                        var passed = angular.equals(actual, expected);
+                        return {
+                            pass: passed,
+                            message: 'Expected ' + actual + (passed ? '' : ' not') + ' to equal ' + expected
+                        }
+                    }
+                }
+            }
+        });
+    });
+
+beforeEach(module('conceptDictionaryApp'));
+
+    describe('controller: ConceptAddController', function() {
+    	  var ctrl, scope, $controller, serverLocales, serviceSpy, classes, datatypes, location;    	  
+
+    	  beforeEach(inject(function($rootScope, $controller) {
+    		  location = {path: function(){}};
+    		  datatypes = {};
+    		  datatypes.results = [{uuid : "8d4a6242-c2cc-11de-8d13-0010c6dffd0f"},
+    		                       {uuid : "8d4a4488-c2cc-11de-8d13-0010c6dffd0f"}]
+    		  
+    		  classes = {};
+    		  classes.results = [{uuid : "8d4a4488-c2cc-11de-8d13-0010c6dffd0f"}]
+    		  
+    		  serverLocales = ["en", "es"];
+    		  serviceSpy = {
+    		    		getEmptyLocaleConceptObject: function() {},
+    		    		getEmptyConceptObject: function() {},
+    		    	    postConcept: function() {}
+    		    	    };    
+    		  
+      		  spyOn(serviceSpy, "getEmptyLocaleConceptObject").and.returnValue({synonyms : [], searchTerms : [], fullname : {}})
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Line is too long.](https://www.codacy.com/app/openmrs/openmrs-owa-conceptdictionary/file/2339254596/issues/source?bid=3196969&fileBranchId=3209439#l52)